addons/survey/models/res_lang.py

```diff
@@ -0,0 +1,24 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command, models, _
+from odoo.exceptions import UserError
+
+
+class ResLang(models.Model):
+    _inherit = "res.lang"
+
+    def write(self, vals):
+        """ When a language is disabled, clear corresponding survey language. """
+        if 'active' in vals and not vals['active']:
+            self.env['survey.user_input'].search([('lang_id', 'in', self.ids)]).lang_id = False
+            surveys = self.env['survey.survey'].search([('lang_ids', 'in', self.ids)])
+            surveys_all_langs = surveys.filtered(lambda survey: not survey.lang_ids)
+            surveys.write({'lang_ids': [Command.unlink(lang.id) for lang in self]})
+            survey_all_langs_after_disabling_lang = surveys.filtered(lambda survey: not survey.lang_ids)
+            if surveys_all_langs != survey_all_langs_after_disabling_lang:
+                impacted_surveys = survey_all_langs_after_disabling_lang - surveys_all_langs
+                raise UserError(
+                    _("Cannot deactivate a language that is currently used by a survey supporting only that language"
+                      " (surveys: %(survey_titles)s).",
+                      survey_titles=', '.join([f'"{survey.title})"' for survey in impacted_surveys])))
```

Comment:
Show names only if read access_right  on all of them.

Reply:
Yes indeed, thanks.